File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"

#include <boost/lexical_cast.hpp>

namespace raindrops {
    std::string convert(int number) {
        std::string output;
        if (number % 3 == 0) output += "Pling";
        if (number % 5 == 0) output += "Plang";
        if (number % 7 == 0) output += "Plong";
        if (output.empty()) return boost::lexical_cast<std::string>(number);
        return output;
    }
}  // namespace raindrops

File: solutions/go/raindrops/1/raindrops.go
package raindrops

import "fmt"

func Convert(number int) string {
    stringToReturn := ""
	if number % 3 == 0 {
        stringToReturn += "Pling"
    }
	if number % 5 == 0 {
        stringToReturn +=  "Plang"
    }
	if number % 7 == 0 {
        stringToReturn +=  "Plong"
    }
	if stringToReturn == "" {
		return fmt.Sprintf("%d", number)
    }
	return stringToReturn
}
